src/firefly/java/edu/caltech/ipac/visualize/plot/FitsRead.java

```diff
@@ -1727,18 +1717,21 @@ public void freeResources() {
 
 
     }
-    static Header cloneHeader(Header header) {
-        // first collect cards from old header
+
+
+    static Header cloneHeaderFrom(Header header) {
         Cursor iter = header.iterator();
-        String cards[] = new String[header.getNumberOfCards()];
-        int i = 0;
+        Header clonedHeader = new Header();
+
         while (iter.hasNext()) {
             HeaderCard card = (HeaderCard) iter.next();
-            cards[i] = card.toString();
-            i++;
+            clonedHeader.addLine(card);
```

Comment:
I think we should make a copy.

`clonedHeader.addLine(card.copy());`
_or_
`clonedHeader.addLine(new HeaderCard(card.getKey(), card.getValue(), card.getComment()));`